lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@rocksfrow no I haven't done any tests I was just assuming that considering the fact that the DSN is client side. Seems you proved the opposite so forget about that. Also I think the PHP constant is not PDO specific as it is defined in the native `pgsql` extension. So let's just stick to `SET NAMES` for now if you don't have a better solution and stop making politics here ;)